files/usr/share/jeos-firstboot/jeos-firstboot-functions

```diff
@@ -0,0 +1,155 @@
+#!/bin/bash
+stty_size() {
+	set -- `stty size`; LINES=$1; COLUMNS=$2
+	# stty size can return zero when not ready or
+	# its a serial console
+	if [ "$COLUMNS" = "0" -o "$LINES" = "0" ]; then
+		LINES=24
+		COLUMNS=80
+	fi
+}
+stty_size
+
+result=
+list=
+password=''
+modules=()
+
+let dh_menu=LINES-15
+let dh_text=LINES-5
+
+if pushd "/usr/share/jeos-firstboot/modules" &>/dev/null; then
+        for module in *; do
+                if [ -f "${module}" ] && source "${module}"; then
+                        modules+=("${module}")
+                fi
+        done
+        popd &>/dev/null
+fi
+
+call_module() {
+	local module="$1"
+	local module_func="$2"
+	module_function="${module}_${module_func}"
+	[ "$(type -t "${module_function}")" = "function" ] || return 1
+	"${module_function}" && true # To not trigger errexit
+	ret=$?
+	[ $ret -eq 0 ] || return $ret
```

Comment:
Just `return $?`

Reply:
I guess so, is just copy and paste from before

Reply:
Previously it was in a loop, so it was needed to break out of it

Reply:
correct